src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs

```diff
@@ -44,20 +47,24 @@ public static class HttpConfigurationExtensions
         {
             Ensure.NotNull(apiFactory, "apiFactory");
 
-            using (var api = apiFactory())
+            HttpRouteCollection routes = config.Routes;
+            ////routePrefix = HttpConfigurationExtensions.RemoveTrailingSlash(routePrefix);
+            if (batchHandler != null && batchHandler.ApiFactory == null)
             {
-                var model = await api.GetModelAsync();
-                model.EnsurePayloadValueConverter();
-                var conventions = CreateRestierRoutingConventions(config, model, apiFactory);
+                batchHandler.ApiFactory = apiFactory;
+                batchHandler.ODataRouteName = routeName;
+                string routeTemplate = string.IsNullOrEmpty(routePrefix) ?
+                    ODataRouteConstants.Batch : (routePrefix + '/' + ODataRouteConstants.Batch);
+                routes.MapHttpBatchRoute(routeName + "Batch", routeTemplate, batchHandler);
+            }
 
-                if (batchHandler != null && batchHandler.ApiFactory == null)
-                {
-                    batchHandler.ApiFactory = apiFactory;
-                }
+            var pathHandler = new DefaultODataPathHandler();
+            ////defaultODataPathHandler.ResolverSetttings = config.GetResolverSettings();
```

Comment:
ResolverSettings is an internal member, must find a way to set it if no other convenient API.